Objects/longobject.c

```diff
@@ -1052,11 +1060,208 @@ _PyLong_AsByteArray(PyLongObject* v,
     return 0;
 
   Overflow:
-    PyErr_SetString(PyExc_OverflowError, "int too big to convert");
+    if (with_exceptions) {
+        PyErr_SetString(PyExc_OverflowError, "int too big to convert");
+    }
     return -1;
 
 }
 
+// Refactored out for readability, not reuse
+static inline int
+_fits_in_n_bits(Py_ssize_t v, Py_ssize_t n)
+{
+    if (n >= (Py_ssize_t)sizeof(Py_ssize_t) * 8) {
+        return 1;
+    }
+    // If all bits above n are the same, we fit.
+    // (Use n-1 if we require the sign bit to be consistent.)
+    Py_ssize_t v_extended = v >> ((int)n - 1);
```

Comment:
In C, right shift on a negative value has implementation-defined behaviour. Do we already depend on compilers doing “the right thing”?

Reply:
We depend on `n` being greater than zero, which is checked before this is called. (And as the comment says, readability, not reuse, so there's no gain by adding a second layer of error checks.)